src/matmul.jl

```diff
@@ -11,7 +10,7 @@ function matmul!(
 end
 
 function matmul!(
-    y::AbstractVector{Float32},
+    y::Vector{Float32},
```

Comment:
This seems relatively strict -- presumably this is to catch non-CUDA vectors?

Reply:
Yeah it was actually the reason, but now everything is using kernels, even the CPU branches with KA, so nothing used `matmul.jl` so it got deleted.